agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/SnippetConfiguration.java

```diff
@@ -20,19 +20,27 @@ public class SnippetConfiguration {
 
   // visible for testing
   static String readSnippet() {
+    ClassLoader classLoader = SnippetConfiguration.class.getClassLoader();
+    String resourceName = "javascript-snippet.txt";
+    InputStream inputStream = classLoader.getResourceAsStream(resourceName);
+    if (inputStream == null) {
+      LOGGER.error("Resource not found: " + resourceName);
+      return "";
+    }
     try {
-      Path path = getSnippetFilePath("javascript-snippet.txt");
-      byte[] bytes = Files.readAllBytes(path);
-      return new String(bytes, StandardCharsets.UTF_8);
-    } catch (IOException | URISyntaxException e) {
+      return toString(inputStream);
+    } catch (IOException e) {
+      // Handle any IO exceptions that occur
       LOGGER.error("Failed to read javascript-snippet file", e);
     }
     return "";
   }
 
-  private static Path getSnippetFilePath(String resourceName) throws URISyntaxException {
-    ClassLoader classLoader = SnippetConfiguration.class.getClassLoader();
-    return Paths.get(classLoader.getResource(resourceName).toURI());
+  private static String toString(InputStream inputStream) throws IOException {
+    try (BufferedReader bufferedReader =
+        new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+      return bufferedReader.lines().collect(Collectors.joining(System.lineSeparator()));
```

Comment:
probably makes sense to always use `\n` instead of `\r\n` on windows server, since this is going to end up on client

```suggestion
      return bufferedReader.lines().collect(Collectors.joining("\n"));
```